packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveSourcePath.ts

```diff
@@ -95,11 +103,62 @@ export async function forceSourceRetrieveSourcePath(explorerPath: vscode.Uri) {
       return;
     }
   }
+
   const commandlet = new SfdxCommandlet(
     new SfdxWorkspaceChecker(),
     new FilePathGatherer(explorerPath),
-    new ForceSourceRetrieveSourcePathExecutor(),
+    useBetaRetrieve(explorerPath)
+      ? new LibraryRetrieveSourcePathExecutor()
+      : new ForceSourceRetrieveSourcePathExecutor(),
     new SourcePathChecker()
   );
   await commandlet.run();
 }
+
+// this supported types logic is temporary until we have a way of generating the metadata type from the path
+// once we have the metadata type we can check to see if it is a toolingsupportedtype from that util
+export function useBetaRetrieve(explorerPath: vscode.Uri): boolean {
+  const filePath = explorerPath.fsPath;
+  const betaDeployRetrieve = sfdxCoreSettings.getBetaDeployRetrieve();
+  const supportedType =
+    path.extname(filePath) === APEX_CLASS_EXTENSION ||
```

Comment:
I found a bug when using the `*-meta.xml` file path in the library so just enabling for the non-meta xml path for now.